linux_os/guide/system/accounts/accounts-restrictions/account_expiration/ensure_shadow_group_empty/bash/shared.sh

```diff
@@ -1,4 +1,4 @@
 # platform = multi_platform_sle,multi_platform_ubuntu
 
-grep '^shadow:[^:]*:[^:]*:[^:]+' /etc/group
-awk -F: '($4 == "<shadow-gid>") { print }' /etc/passwd
+sed -ri 's/(^shadow:[^:]*:[^:]*:)([^:]+$)/\1/' /etc/group
+awk -F: -v GID="$(awk -F: '($1=="shadow") {print $3}' /etc/group)" '($4==GID) {print $1}' /etc/passwd
```

Comment:
I believe this `awk` command is very useful for checking, but it seems unnecessary in the remediation since it is only showing if any user has the `shadow` as primary group. Would it make sense to change the primary group of these users, if any is found? Otherwise we should include a warning in the `rule.yml` informing this situation.

Reply:
Also a test for this scenario would be interesting.

Reply:
> I believe this `awk` command is very useful for checking, but it seems unnecessary in the remediation since it is only showing if any user has the `shadow` as primary group. Would it make sense to change the primary group of these users, if any is found? Otherwise we should include a warning in the `rule.yml` informing this situation.

I think changing any users primary group is too disruptive.
I can remove the line.

Reply:
> Also a test for this scenario would be interesting.

not sure I understood what scenario you are referring to.
We already have a test for it: `shadow_user_member.fail.sh`  for users with primary group shadow

Reply:
I've removed the line you mentioned